### ISSUE TITLE
[Bug]: Flatpak apps won't start

### ISSUE THREAD

**Issue by ghost** [flatpak/flatpak#4637]
### Checklist

- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a bug that matches the one I want to file, without success.
- [x] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] If this is an issue with a particular app, I have tried filing it in the appropriate issue tracker for the app (e.g. under https://github.com/flathub/) and determined that it is an issue with Flatpak itself.

### Flatpak version

1.12.2

### What Linux distribution are you using?

Fedora Linux

### Linux distribution version

35

### What architecture are you using?

x86_64

### How to reproduce

1 `Open any flatpak app`
2 `Wait`
3 `Nothing happens`

### Expected Behavior

To run flatpak apps

### Actual Behavior

Nothing happens, even looked at journalctl , but no errors are showed

with verbose, no error appears.
```
F: No installations directory in /etc/flatpak/installations.d. Skipping
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/ivan/.local/share/flatpak
F: Opening user flatpak installation at path /home/ivan/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/ivan/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: /var/lib/flatpak/runtime/org.freedesktop.Platform/x86_64/21.08/940b0728d1b7d18b9d3f557ae44e7c37b27febcb04f93a7cc0ebb886ba2cfa37/files/lib32 does not exist
F: Allocated instance id 409466375
F: Add defaults in dir /org/telegram/desktop/
F: Add locks in dir /org/telegram/desktop/
F: Allowing host-fs access
```

At reboot works, sometimes

### Additional Information

When i reboot machine, it will run
After another reboot it will stop again

**Comment by h0nus** [comment]
UPDATE: on latest updates it seems to not even start after a reboot..
i wonder why

**Comment by jalberto** [comment]
Same here, since upgraded to Fedora 35 flatpaks are not able to run and there is no message output when using `-vv`

**Comment by brianredbeard** [comment]
Running into the same thing on Fedora 33, only I can confirm that applications are _starting_ and _running_, they're just not correctly presenting themselves to the window manager (if I had to guess, it's some breakdown between flatpak, dbus, & [xdg-desktop-portal](https://github.com/flatpak/xdg-desktop-portal)).

I start out making sure Signal isn't running:

```bash
[~]$ flatpak ps 
[~]$
```

I then start it through GNOME (e.g. `<SUPER> Signal <ENTER>`)

```bash
[~]$ flatpak ps 
Instance   PID    Application            Runtime
3870315324 945533 org.signal.Signal      org.freedesktop.Platform
2727757526 945507 org.signal.Signal      org.freedesktop.Platform
[~]$ pgrep -f -a signal 
945507 bwrap --args 42 signal
945518 bwrap --args 42 signal
945519 /app/Signal/signal-desktop
945528 /app/Signal/signal-desktop --type=zygote --no-zygote-sandbox
945530 /app/Signal/chrome-sandbox /app/Signal/signal-desktop --type=zygote
945533 bwrap --args 44 /app/bin/zypak-helper child - /app/Signal/signal-desktop --type=zygote
945546 bwrap --args 44 /app/bin/zypak-helper child - /app/Signal/signal-desktop --type=zygote
945547 /app/Signal/signal-desktop --type=zygote
945582 /app/Signal/signal-desktop --type=gpu-process --field-trial-handle=17111136317911695144,16301358051019578441,131072 --disable-features=SpareRendererForSitePerProcess --user-data-dir=/home/bharrington/.var/app/org.signal.Signal/config/Signal --gpu-preferences=UAAAAAAAAAAgAAAIAAAAAAAAAAAAAAAAAABgAAAAAAAwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABgAAAAAAAAAGAAAAAAAAAAIAAAAAAAAAAgAAAAAAAAACAAAAAAAAAA= --shared-files
945592 /app/Signal/signal-desktop --type=utility --utility-sub-type=network.mojom.NetworkService --field-trial-handle=17111136317911695144,16301358051019578441,131072 --disable-features=SpareRendererForSitePerProcess --lang=en-US --service-sandbox-type=none --user-data-dir=/home/bharrington/.var/app/org.signal.Signal/config/Signal --shared-files=v8_context_snapshot_data:100
945615 /app/Signal/signal-desktop --type=renderer --user-data-dir=/home/bharrington/.var/app/org.signal.Signal/config/Signal --app-path=/app/Signal/resources/app.asar --no-sandbox --no-zygote --field-trial-handle=17111136317911695144,16301358051019578441,131072 --disable-features=SpareRendererForSitePerProcess --lang=en-US --num-raster-threads=4 --enable-main-frame-before-activation --renderer-client-id=4 --no-v8-untrusted-code-mitigations --shared-files=v8_context_snapshot_data:100
945668 /app/Signal/signal-desktop --type=utility --utility-sub-type=audio.mojom.AudioService --field-trial-handle=17111136317911695144,16301358051019578441,131072 --disable-features=SpareRendererForSitePerProcess --lang=en-US --service-sandbox-type=none --user-data-dir=/home/bharrington/.var/app/org.signal.Signal/config/Signal --shared-files=v8_context_snapshot_data:100
```

**Comment by h0nus** [comment]
Before Fedora, i was on ubuntu and got the same issue.
While running into this issue, the OS was very very unstable.
Like, nautilus was opening after 40 second from launching it.
With some research around internet, i found that was something related to `xdg-desktop-portal`.

Tried to remove it and everything was working smooth after i've removed it.
On ubuntu, removing it, it would remove flatpak too.
So i think it's somewhat related.

**Comment by JavkhlanK** [comment]
I'm getting the same issue now with Firefox Beta v104.0b8 (tried using stable before, no results), and Android Studio v2021.2.1.16.

I initially made a report to Mozilla's Bugzilla as by then only Firefox wouldn't launch. Now it's Android Studio too, and there seems to be no reason, in neither stdout/stderr when running with `flatpak run`, or `journalctl`.

Bugzilla report: https://bugzilla.mozilla.org/show_bug.cgi?id=1773546  
It contains some screenshots of the output I'm getting, pretty much nothing abnormal.

I also doubt that they're running 100 % normally, as the processes did get started, but were pretty much doing nothing (used GNOME System Monitor and saw 0 % CPU usage all the time)

Using Fedora Linux Workstation Edition 36, flatpak v1.12.7.

**Comment by ErikLentz** [comment]
Same exact problem. Fedora 39 and Flatpak 1.15.6. Sometimes it works after reboots, sometimes not. I can find the flatpak processes running, and a verbose run of an app looks much like the above, but the app doesn't appear.

**Comment by glennrice** [comment]
Getting this as well on Fedora 39 and 40.

After a reboot, or sometimes two, Flatpak apps open again. I thought it was just the electron ones at first but appear to be all.

Tried a run verbose and got the following:

```
$ flatpak run -v org.kde.marknote
F: No installations directory in /etc/flatpak/installations.d. Skipping
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/glenn/.local/share/flatpak
F: Opening user flatpak installation at path /home/glenn/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: Skipping parental controls check for app/org.kde.marknote/x86_64/stable since parental controls are disabled globally
F: Opening user flatpak installation at path /home/glenn/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: /var/lib/flatpak/runtime/org.kde.Platform/x86_64/6.6/a34c61c13fb4dc31bd219551c18f7a60e6c00c7e5f64b5091bcdba2ac3615be6/files/lib32 does not exist
F: Cleaning up unused container id 2246480401
F: Cleaning up per-app-ID state for org.kde.marknote
F: Allocated instance id 1607879234
F: Add defaults in dir /org/kde/marknote/
F: Add locks in dir /org/kde/marknote/
F: Disallowing network access
F: Allowing dri access
F: Allowing wayland accessnn@glenn-ThinkPad:~$ flatpak run -v org.kde.marknote
F: No installations directory in /etc/flatpak/installations.d. Skipping
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/glenn/.local/share/flatpak
F: Opening user flatpak installation at path /home/glenn/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: Skipping parental controls check for app/org.kde.marknote/x86_64/stable since parental controls are disabled globally
F: Opening user flatpak installation at path /home/glenn/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: /var/lib/flatpak/runtime/org.kde.Platform/x86_64/6.6/a34c61c13fb4dc31bd219551c18f7a60e6c00c7e5f64b5091bcdba2ac3615be6/files/lib32 does not exist
F: Cleaning up unused container id 2246480401
F: Cleaning up per-app-ID state for org.kde.marknote
F: Allocated instance id 1607879234
F: Add defaults in dir /org/kde/marknote/
F: Add locks in dir /org/kde/marknote/
F: Disallowing network access
F: Allowing dri access
F: Allowing wayland access
```

Notice the last line is about Wayland...

**Comment by ppvnf** [comment]
in my case initially after the session starts flatpaks work and then they stop being able to be launched eventually. I don't have to reboot for them to work tho, only logging out and in is enough

**Comment by Searge** [comment]
Fedora 40, Flatpak 1.15.10

The same issue:

```
❯ flatpak run -v md.obsidian.Obsidian
F: No installations directory in /etc/flatpak/installations.d. Skipping
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/searge/.local/share/flatpak
F: Opening user flatpak installation at path /home/searge/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: Opening user flatpak installation at path /home/searge/.local/share/flatpak
F: Opening system flatpak installation at path /var/lib/flatpak
F: /var/lib/flatpak/runtime/org.freedesktop.Platform/x86_64/24.08/4032cef5b0b698b46df684fc68527252cb7350f39bd90b25bb0c79766911de23/files/lib32 does not exist
F: Cleaning up unused container id 1832069502
F: Allocated instance id 4500041
F: Add defaults in dir /md/obsidian/Obsidian/
F: Add locks in dir /md/obsidian/Obsidian/
F: Allowing dri access
F: Allowing homedir access
F: Allowing x11 access
F: Allowing pulseaudio access
F: Pulseaudio user configuration file '/home/searge/.config/pulse/client.conf': Error opening file /home/searge/.config/pulse/client.conf: No such file or directory
```

I have to reboot or log out to make it work again.

**Comment by glennrice** [comment]
Updated to Flatpak 1.16.0 and Plasma 6.2.5 - unfortunately, the same issue.

**Comment by SpamixOfficial** [comment]
Happening to me as-well, on latest version of everything. Happened after a reboot and rebooting again and updating haven't helped

**Comment by glennrice** [comment]
I notice that this appears to be a timing thing! Logging in and opening a Flatpak app straight away usually fails - process starts but never a window.

Waiting a minute or so after logging in before opening anything and usually all is OK. A Flatpak app opens straight away.

Still trying to gather evidence for this. Any ideas of where to look appreciated.

**Comment by smcv** [comment]
> I don't see how the original reporter disappearing is relevant, given a lot of people over the years have reported facing the same, even very recently.

"App won't start" is very much a symptom, not a root cause. Even if we can identify and solve whatever problem is causing this symptom to happen in 2025 for you, because the original reporter has disappeared, we can never know whether it's really the same problem that was causing this symptom in 2021 for them (or for any of the people who have experienced a similar symptom in the intervening period).

> I can also confirm that the workaround of waiting a bit better logging in and launching an app mentioned above, seems to work.

I'm having trouble parsing "waiting a bit better logging in". Do you mean the workaround is waiting a bit after logging in, so that: log out, log in, immediately launch a Flatpak app → fails; log out, log in, wait approx. 60 seconds, launch a Flatpak app → success?

If yes, then this appears to be true for you and for @glennrice, but not necessarily for everyone who has commented here. A separate issue report with a title like "Flatpak apps won't start immediately after login, but work OK after 60 seconds" would capture that scenario more specifically than just "Flatpak apps won't start", and seems more likely to yield a solution.

Conversely, @ppvnf seems to have had the opposite, where Flatpak apps *will* start at some earlier time, but stop working at some later time. This suggests that @glennrice and @ppvnf are/were probably experiencing different root causes for a superficially similar symptom, such that solving one of those root causes will not necessarily solve the other.

> If you could please share what kind of logs are needed here, I can try to provide them.

Unfortunately, if I knew precisely what would identify a root cause, I'd be halfway to solving it - which I am not.

As a minimum, the sort of system details that the issue template asks for are always useful information. I'm assuming that you are not using Flatpak 1.12.2 on Fedora 35 like the original reporter of this issue was!

Being able to reproduce this in as simple, reproducible or debuggable a scenario as possible is always useful. For example if this happens for every app, then an app that doesn't have any particularly interesting or unusual permissions associated with it, like `org.gnome.Recipes` or `org.gnome.Nibbles`, is going to be the most interesting. Or, if this happens for only a subset of apps, what we would ideally want to find is a pair of apps, one with the problem and one without, that have sandbox parameters as similar as possible.

Similarly if this can be reproduced in multiple environments, it will be more likely to be useful to have debug logs, steps to reproduce etc. from a relatively "normal" desktop environment (an ordinary distro installation of GNOME or KDE Plasma or something, on Fedora or Debian or something) than a build-your-own environment like Sway on Gentoo where every user's system is unique.

`flatpak run -v -v` often provides useful information.

If we suspect a race condition or timing issue, then some information about the relative timing of the lines of output from `flatpak run -v -v` (for example annotating it with any particularly long pauses) would be useful. For instance if it logs "Frobnicating widgets" and then does not produce any output for a long time, then that's a clue that it's taking an unexpectedly long time to either frobnicate the widgets, or do a step that comes after that and has no specific debug messages.

Depending what is happening, `strace` might also provide clues.

If you are seeing a transition from one state to another - for example if you log in at 18:00 and Flatpak apps can't start, but by 18:01 they are working OK - then it would be useful to see whatever messages appear in the systemd Journal during the interval in which the state transition occurred. For example, that might narrow it down to something more like "Flatpak apps can't start until after `foobar.service` has finished startup".

**Comment by glennrice** [comment]
What I meant by timing issue, is if I logon and then wait around 3 minutes, then open a flatpak it usually works. 
Opening one straight after logon fails. The interesting exception to this is any flatpaks set to autostart in KDE always open, which is strange not sure what the difference is there.

Thanks for the tip, re: `flatpak run -v -v`    I'll see if that shows up anything over the next week or so.

**Comment by bbhtt** [comment]
Closing this, 6 years is long enough to serve as a catch-all issue where everyone is posting their own issues and is not actionable.
